### DISCUSSION TITLE
Predictable preview url for monorepo

### DISCUSSION THREAD

**Discussion by ramiel** [vercel/vercel#6646] [Help] [answered]
I have a project made of several websites in a monorepo. In order to have preview environment working I was using the git url to connect one website to the other in preview. For example if website A was deployed to `websiteA-git-commit.vercel.app` I knew I could point to website B containing the same changeset pointing to `websiteB-git-commit.vercel.app`.

Stated that this was already sub-optimal, it looks like today this is not possible anymore. The preview url for the git branch has a `r-xyz123` part that is different for each website and I cannot connect the same deployment through several website. As I already said somewhere else, visiting websites independently is not possible sometimes, usually because of authentication issue (that may rely on domain name for the cookie).

Also, this is against what is written in documentation about "[memorable preview links](https://vercel.com/support/articles/why-do-my-vercel-deployments-have-multiple-domains?query=preview%20url#memorable-preview-links)".

Vercel monorepo support is not a pleasant experience for developers as it could be. How can we solve this situation?

**Comment by leerob** [comment, marked as answer]
Hey there! We've just released new documentation describing our automatic URLs: https://vercel.com/docs/deployments/automatic-urls

Does this solve your question?
